src/sagas/walletConnect.js

```diff
@@ -0,0 +1,424 @@
+    console.log('Captured error on signMessage: ', error);
+    yield put(onExceptionCaptured(error));
+  }
+}
+
+export function* onWalletReset() {
+  const { web3wallet } = yield select((state) => state.walletConnect);
+  if (!web3wallet) {
+    // Do nothing, wallet connect might not have been initialized yet
+    return;
+  }
+
+  yield call(clearSessions);
+}
+
+export function* onSessionProposal(action) {
+  const { id, params } = action.payload;
+  const { web3wallet } = yield select((state) => state.walletConnect);
+
+  const wallet = yield select((state) => state.wallet);
+  const addresses = wallet.newAddresses;
+  const firstAddress = addresses[0].address;
+
+  const data = {
+    icon: get(params, 'proposer.metadata.icons[0]', null),
+    proposer: get(params, 'proposer.metadata.name', ''),
+    url: get(params, 'proposer.metadata.url', ''),
+    description: get(params, 'proposer.metadata.description', ''),
+    requiredNamespaces: get(params, 'requiredNamespaces', []),
+  };
+
+  const onAcceptAction = { type: 'WALLET_CONNECT_ACCEPT' };
+  const onRejectAction = { type: 'WALLET_CONNECT_REJECT' };
+
+  yield put(setWalletConnectModal({
+    show: true,
+    type: WalletConnectModalTypes.CONNECT,
+    data,
+    onAcceptAction,
+    onRejectAction,
+  }));
+
+  const { accept } = yield race({
+    accept: take(onAcceptAction.type),
+    reject: take(onRejectAction.type),
+  });
+
+  if (!accept) {
+    yield call(() => web3wallet.rejectSession({
+      id: params.id,
+      reason: {
+        code: USER_REJECTED,
+        message: 'User rejected the session',
+      },
+    }));
+  }
+
+  try {
+    yield call(() => web3wallet.approveSession({
+      id,
+      relayProtocol: params.relays[0].protocol,
+      namespaces: {
+        hathor: {
+          accounts: [`hathor:${NETWORK}:${firstAddress}`],
+          chains: [`hathor:${NETWORK}`],
+          events: AVAILABLE_EVENTS,
+          methods: AVAILABLE_METHODS,
+        },
+      },
+    }));
+
+    yield call(refreshActiveSessions);
+  } catch(error) {
+    console.error('Error on sessionProposal: ', error);
+    yield put(onExceptionCaptured(error));
+  }
+}
+
+export function* onQrCodeRead(action) {
+  const { web3wallet, core } = yield select((state) => state.walletConnect);
+
+  if (!web3wallet) {
+    throw new Error('Wallet connect instance is new and QRCode was read');
+  }
+
+  const { payload } = action;
+
+  try {
+    yield call(() => core.pairing.pair({ uri: payload }));
+  } catch(error) {
+    console.error('Error pairing with QrCode: ', error); 
+    yield put(onExceptionCaptured(error));
+  }
+}
+
+export function* featureToggleUpdateListener() {
+  while (true) {
+    const oldWalletConnectEnabled = yield call(isWalletConnectEnabled);
+    yield take('FEATURE_TOGGLE_UPDATED');
+    const newWalletConnectEnabled = yield call(isWalletConnectEnabled);
+
+    if (oldWalletConnectEnabled && !newWalletConnectEnabled) {
+      yield put({ type: 'WC_SHUTDOWN' });
```

Comment:
Are you using this `WC_SHUTDOWN` somewhere?

Reply:
Fixed in be7d9316b162603ffd749fee6be680188c4ad319